hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/web/TestWebHDFS.java

```diff
@@ -2230,6 +2230,25 @@ public void testLinkTarget() throws Exception {
     }
   }
 
+  @Test
+  public void testFileLinkStatus() throws Exception {
+    final Configuration conf = WebHdfsTestUtil.createConf();
+    cluster = new MiniDFSCluster.Builder(conf).numDataNodes(3).build();
+    cluster.waitActive();
+
+    final WebHdfsFileSystem webHdfs = WebHdfsTestUtil.getWebHdfsFileSystem(conf,
+        WebHdfsConstants.WEBHDFS_SCHEME);
+    // Symbolic link
+    Path root = new Path("/webHdfsTest/");
+    Path file = new Path(root, "file");
+    FileSystemTestHelper.createFile(webHdfs, file);
+
+    Path linkToFile = new Path(root, "linkToFile");
+
+    webHdfs.createSymlink(file, linkToFile, false);
+    assertTrue(webHdfs.getFileLinkStatus(linkToFile).isSymlink());
+  }
```

Comment:
Can you also test that the correct result is returned when the path is not a symLink.

Reply:
Thanks for your suggestion, I'll make the required code changes promptly